internal/encoding/merge/parquet.go

```diff
@@ -0,0 +1,326 @@
+		col.SchemaElement.LogicalType = parquet.NewLogicalType()
+		col.SchemaElement.LogicalType.INTEGER = &parquet.IntType{BitWidth: 32, IsSigned: true}
+		col.SchemaElement.ConvertedType = parquet.ConvertedTypePtr(parquet.ConvertedType_INT_32)
+		return col, optional(uintHandler(32)), nil
+	case "uint32":
+		col.SchemaElement.Type = parquet.TypePtr(parquet.Type_INT32)
+		col.SchemaElement.LogicalType = parquet.NewLogicalType()
+		col.SchemaElement.LogicalType.INTEGER = &parquet.IntType{BitWidth: 32, IsSigned: false}
+		col.SchemaElement.ConvertedType = parquet.ConvertedTypePtr(parquet.ConvertedType_UINT_32)
+		return col, optional(uintHandler(32)), nil
+	case "int64":
+		col.SchemaElement.Type = parquet.TypePtr(parquet.Type_INT64)
+		col.SchemaElement.LogicalType = parquet.NewLogicalType()
+		col.SchemaElement.LogicalType.INTEGER = &parquet.IntType{BitWidth: 64, IsSigned: true}
+		col.SchemaElement.ConvertedType = parquet.ConvertedTypePtr(parquet.ConvertedType_INT_64)
+		return col, optional(intHandler(64)), nil
+	case "uint64":
+		col.SchemaElement.Type = parquet.TypePtr(parquet.Type_INT64)
+		col.SchemaElement.LogicalType = parquet.NewLogicalType()
+		col.SchemaElement.LogicalType.INTEGER = &parquet.IntType{BitWidth: 64, IsSigned: false}
+		col.SchemaElement.ConvertedType = parquet.ConvertedTypePtr(parquet.ConvertedType_UINT_64)
+		return col, optional(uintHandler(64)), nil
+	case "float64":
+		col.SchemaElement.Type = parquet.TypePtr(parquet.Type_FLOAT)
+		return col, optional(floatHandler), nil
+	case "double":
+		col.SchemaElement.Type = parquet.TypePtr(parquet.Type_DOUBLE)
+		return col, optional(doubleHandler), nil
+	case "int":
+		col.SchemaElement.Type = parquet.TypePtr(parquet.Type_INT64)
+		col.SchemaElement.LogicalType = parquet.NewLogicalType()
+		col.SchemaElement.LogicalType.INTEGER = &parquet.IntType{BitWidth: 64, IsSigned: true}
+		col.SchemaElement.ConvertedType = parquet.ConvertedTypePtr(parquet.ConvertedType_INT_64)
+		return col, optional(intHandler(64)), nil
+	case "json":
+		col.SchemaElement.Type = parquet.TypePtr(parquet.Type_BYTE_ARRAY)
+		col.SchemaElement.LogicalType = parquet.NewLogicalType()
+		col.SchemaElement.LogicalType.JSON = &parquet.JsonType{}
+		col.SchemaElement.ConvertedType = parquet.ConvertedTypePtr(parquet.ConvertedType_JSON)
+		return col, optional(byteArrayHandler), nil
+	default:
+		return nil, nil, fmt.Errorf("toparquet:unsupported type %q", typ)
+	}
+}
+
+func byteArrayHandler(s interface{}) (interface{}, error) {
+	switch v := s.(type){
+	case []byte:
+		return v, nil
+	case string:
+		return []byte(v), nil
+	case json.RawMessage:
+		return []byte(v), nil
+	default:
+		return []byte(fmt.Sprintf("%v", s)), nil
+	}
+}
+
+func booleanHandler(s interface{}) (interface{}, error) {
+	switch v := s.(type){
+	case bool:
+		return v, nil
+	case string:
+		return strconv.ParseBool(v)
+	default:
+		return nil, fmt.Errorf("toparquet:Unable to parse as boolean %d", s)
```

Comment:
Add space after 'toparquet: ' and make sure the message is in lower case (for all error messages)